ambry-router/src/main/java/com.github.ambry.router/SingleKeyManagementService.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.github.ambry.account.Account;
+import com.github.ambry.account.Container;
+import com.github.ambry.config.KMSConfig;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import org.bouncycastle.util.encoders.Hex;
+
+
+/**
+ * Single {@link KeyManagementService} which returns a default key {@link SecretKeySpec} for any
+ * {@link #getKey(short, short)} calls but a different random key for {@link #getRandomKey()} calls.
+ */
+public class SingleKeyManagementService implements KeyManagementService<SecretKeySpec> {
+  private final SecretKeySpec secretKeySpec;
+  private volatile boolean enabled = true;
+  private final KeyGenerator keyGen;
+  private final String keyGenAlgo;
+
+  SingleKeyManagementService(KMSConfig config, String defaultKey) throws GeneralSecurityException {
```

Comment:
This key is never overridden, so it is more than the default key, maybe the "containerKey". Also add javadocs about the format of this key (like that it needs to be a hex string).